http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/ToolDAO.scala

```diff
@@ -8,3 +9,9 @@ import scala.concurrent.Future
 trait ToolDAO {
   def isProxyAvailable(googleProject: GoogleProject, clusterName: ClusterName): Future[Boolean]
 }
+
+object ToolDAO {
```

Comment:
Ok I see - the individual DAOs no longer directly extend `ToolDAO`, we're now adapting them to `ToolDAO` via these methods.

I was questioning to myself whether we could use a typeclass for this -- define implicit converters from JupyterDAO/WelderDAO/RStudioDAO -> ToolDAO and summon them like `ToolDAO[T].isProxyAvailable`. But I'm not sure if that's exactly right for our case, where we're passing the ToolDAOs in a dynamic Map.

Anyway I'm fine with this as is, just thinking out loud.

Reply:
yea, I think that's a good idea...initially, I just really didn't want say `HttpJupyterDAO` to extend from both `JupyterDAO` and `ToolDAO`.